Core/TimberApi/ToolGroupSystem/ToolGroupFactoryService.cs

```diff
@@ -0,0 +1,31 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+
+namespace TimberApi.ToolGroupSystem
+{
+    public class ToolGroupFactoryService
+    {
+        private readonly ImmutableDictionary<string, IToolGroupFactory> _toolGroupFactories;
+
+        public ToolGroupFactoryService(IEnumerable<IToolGroupFactory> toolFactories)
+        {
+            _toolGroupFactories = toolFactories.ToImmutableDictionary(factory => factory.Id.ToLower());
+        }
+
+        public IToolGroupFactory Get(string factoryId)
+        {
+            if(! TryGet(factoryId, out var toolFactory))
```

Comment:
Why the wrapper method `TryGet`, when calling just `_toolGroupFactories.TryGetValue` does exactly the same?

Reply:
TryGet is public method to be used for users